jinahub/indexers/SimpleIndexer/simple_indexer.py

```diff
@@ -97,6 +97,45 @@ def search(
         )
         self._flush = False
 
+    @requests(on='/delete')
+    def delete(self, docs: DocumentArray, parameters: Optional[Dict] = {}, **kwargs):
+        """Delete entries from the index by id
+
+        :param docs: the documents to delete
+        :param parameters: parameters to the request
+        """
+        if docs is None:
+            return
+        traversal_paths = parameters.get(
+            'traversal_paths', self.default_traversal_paths
+        )
+        flat_docs = docs.traverse_flat(traversal_paths)
+        delete_docs_ids = set(flat_docs.get_attributes('id'))
+        for idx in reversed(range(len(self._docs))):
+            if self._docs[idx].id in delete_docs_ids:
+                del self._docs[idx]
```

Comment:
I think the  `__delete__` method in `DAM` can use `doc.id` directly
```suggestion
        for doc in flat_docs:
            if doc.id is None: continue
            deleted_doc = self._docs.get(doc.id, None)
            if deleted_doc:
                del self._docs[doc.id]
        
```

Reply:
can't you do:
```python
        delete_docs_ids = set(flat_docs.get_attributes('id'))
        for idx in delete_docs_ids:
            if idx in self._docs:
                del self._docs[idx]
```
?

Reply:
oh, looks much better

Reply:
what is the time complexity of `idx in self._docs` ? is it O(1)? If so then it is better. @alaeddine-13